src/test/java/com/github/slem1/await/HttpServiceTest.java

```diff
@@ -22,7 +22,8 @@ public void shouldConnectAndGet200() throws IOException, ServiceUnavailableExcep
         HttpURLConnection urlConnection = Mockito.mock(HttpURLConnection.class);
         when(urlConnection.getResponseCode()).thenReturn(200);
         when(url.openConnection()).thenReturn(urlConnection);
-        HttpService httpService = new HttpService(url, 200);
+        when(url.getProtocol()).thenReturn("http");
```

Comment:
You have fixed the existing test cases but you haven't add any new test to cover the new feature code. Can you fix it please.

Reply:
Done.